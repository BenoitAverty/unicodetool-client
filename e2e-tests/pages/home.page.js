const Page = require('./page')

const homePage = Object.create(Page, {
  url: { value: '/' },
  pageHeading: { value: '#page-title' },
  searchField: { value: '.UnicodeSearchField > input' },
  searchResult: { value: '.searchResult' },
  searchResultItems: { value: '.searchResult .CodepointSummary' },
})

module.exports = homePage
