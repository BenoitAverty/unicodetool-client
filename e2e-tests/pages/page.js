/** Stateless prototype object required for all pages **/

function Page() {
}

Page.prototype.open = function (path) {
    browser.url('/' + path);
};

module.exports = new Page();