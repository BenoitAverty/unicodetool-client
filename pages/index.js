import withRedux from 'next-redux-wrapper'

import makeStore from '../redux/store'
import Dispatcher from '../components/Dispatcher'

const Index = props => (
  <div>
    <p>Next + Redux + Redux-Cycles : {props.count}</p>
    <Dispatcher />
  </div>
)

export default withRedux(makeStore, state => state)(Index)
