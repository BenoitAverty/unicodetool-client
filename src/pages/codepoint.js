import React from 'react'
import fetch from 'isomorphic-fetch'
import withRedux from 'next-redux-wrapper'

import makeStore from '../redux/store'
import { codepointLookupSuccess } from '../redux/actions'
import { codepointLookupQuery } from '../redux/cycles/graphqlRequests'
import Layout from '../components/Layout'
import CodepointSummary from '../components/CodepointSummary'
import { codepointById } from '../redux/reducer'

class CodepointPage extends React.Component {
  static async getInitialProps({ store, query }) {
    return fetch('https://unicodetool-api.now.sh/graphql', {
      method: 'POST',
      body: JSON.stringify({
        query: codepointLookupQuery,
        variables: {
          value: query.cp,
        },
      }),
    })
      .then(response => response.json())
      .then(body => {
        store.dispatch(codepointLookupSuccess(body.data.codepoint))
        return {}
      })
  }

  render() {
    return (
      <Layout>
        <p>{this.props.testProp}</p>
        <CodepointSummary codepoint={this.props.codepoint} />
      </Layout>
    )
  }
}

export const withoutRedux = CodepointPage

const mapStateToProps = (state, { url }) => ({
  codepoint: codepointById(state, url.query.cp),
})

export default withRedux(makeStore, mapStateToProps)(CodepointPage)
