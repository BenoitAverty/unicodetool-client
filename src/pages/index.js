import withRedux from 'next-redux-wrapper'

import makeStore from '../redux/store'

import Layout from '../components/Layout'
import Dispatcher from '../components/Dispatcher'
import CounterWarning from '../components/CounterWarning'

const Index = props => (
  <Layout>
    <div>
      <CounterWarning count={props.count} maxCount={5} />
      <p>Next + Redux + Redux-Cycles : {props.count}</p>
      <Dispatcher />
    </div>
  </Layout>
)

export default withRedux(makeStore, state => state)(Index)
