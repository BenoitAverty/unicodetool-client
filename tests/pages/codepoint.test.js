/* eslint-env jest */
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { withoutRedux as CodepointPage } from '../../src/pages/codepoint'

configure({ adapter: new Adapter() })

describe('Codepoint page', () => {
  describe('With an existing codepoint', () => {
    it('matches snapshot', () => {
      const page = shallow(<CodepointPage codepoint={{}} />)

      expect(toJson(page)).toMatchSnapshot()
    })
  })
})
