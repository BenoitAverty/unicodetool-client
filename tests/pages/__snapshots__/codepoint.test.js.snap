// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Codepoint page With an existing codepoint matches snapshot 1`] = `
<Layout>
  <p />
  <CodepointSummary
    codepoint={Object {}}
  />
</Layout>
`;
